### PR TITLE
Fixed coursId validation

### LINKED ISSUE #255: Fix course isExist check when updating and creating student group

### PR DESCRIPTION
**Pull request by filinskyi**
fixed, course is exist check when creating and updating student group and updated unit test

### CODE REVIEW

**Review by SnipeFrost — APPROVED** [review]

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs [hunk 1]
```diff
@@ -65,9 +65,20 @@ public async Task CreateStudentGroup()
                 MentorIds = new List<long>()
             };
 
+            var withNotExistCourseIdStudentGroup = new CreateStudentGroupDto()
```
Comment (SerjKocherga) [inline CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs]: Let's name it studentGroupWithoutCourseId.

File: CharlieBackend.Business/Services/StudentGroupService.cs [hunk 2]
```diff
@@ -87,7 +92,7 @@ public async Task<Result<StudentGroupDto>> CreateStudentGroupAsync(CreateStudent
                 return Result<StudentGroupDto>.GetSuccess(_mapper.Map<StudentGroupDto>(studentGroup));
 
             }
-            catch(Exception ex)
+            catch (Exception ex)
             {
                 _unitOfWork.Rollback();
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/StudentGroupService.cs:98]: Please, remove "Console.WriteLine(ex.Message)" below.

File: CharlieBackend.Business/Services/StudentGroupService.cs [hunk 3]
```diff
@@ -155,11 +160,12 @@ public async Task<Result<StudentGroupDto>> UpdateStudentGroupAsync(long groupId,
                     foundStudentGroup.FinishDate = (DateTime?)(updatedStudentGroupDto.FinishDate) ?? foundStudentGroup.FinishDate;
                 }
 
-                if (updatedStudentGroupDto.CourseId != 0)
+                if (!await _unitOfWork.CourseRepository.IsEntityExistAsync(updatedStudentGroupDto.CourseId))
                 {
-                    foundStudentGroup.Course = await _unitOfWork.CourseRepository.GetByIdAsync(updatedStudentGroupDto.CourseId);
+                    return Result<StudentGroupDto>.GetError(ErrorCode.ValidationError, "CourseId isn't exist");
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/StudentGroupService.cs]: "CourseId does not exist".

**Review by SerjKocherga — CHANGES_REQUESTED** [review]
Need some changes:

**Review by SerjKocherga — APPROVED** [review]

**Review by sober607 — APPROVED** [review]

**Review by BohdanBBG — APPROVED** [review]

**Review by Hlib-mats — APPROVED** [review]

**Review by whereismyroot — APPROVED** [review]

**Pull request merged**